palantirnet/search_api_federated_solr
# PL-31: Move search app site name configuration to configuration form.

https://palantir.atlassian.net/browse/PL-31

## Description

Remove the alter hook, `search_api_federated_solr_form_search_api_federated_solr_search_app_settings_alter()` and add this logic to the rest of the search app configuration form.

This provides the option to search only on the current site by default. When there is a site_name field sent to the index, the search app configuration form will provide an option to pre-check the current site. If this is checked, the site visitor can still uncheck that option and see results from all sites.

## To Test

- Install a site and set up an index
- Edit the index and make sure there is **no** `site_name` field, example:  /admin/config/search/search-api/index/federated_search_index/fields
- Go to the search app configuration page, /admin/config/search-api-federated-solr/search-app/settings
- In the inspector, look at the value for the hidden `site_name_property` input field
- The value should be `true`
- There should be no `Set the "site name" facet to this site` option.

![site_name_facet_option](https://user-images.githubusercontent.com/397902/50912366-9d64f580-13f7-11e9-911c-5ee59dd9f1c0.jpg)

- Go to the homepage
- Search for a term
- Confirm that the site name facet is not preselected

- Edit the index and **add** a `site_name` field
- Go to the search app configuration page
- In the inspector, look at the value for the hidden `site_name_property` input field
- The value should be blank
- There should be a `Set the "site name" facet to this site` option.
- Check this option

![site_name_facet_option](https://user-images.githubusercontent.com/397902/50912360-99d16e80-13f7-11e9-93da-13d35a1a7a4b.jpg)

- Go to the homepage
- Search for a term
- Confirm that the site name facet is preselected
